governator-test-junit/src/main/java/com/netflix/governator/guice/test/AnnotationBasedTestInjectorManager.java

```diff
@@ -0,0 +1,210 @@
+        injector = createInjector(modulesForTestClass, overrideModules);
+    }
+
+    /**
+     * Injects dependencies into the provided test object.
+     */
+    public void prepareTestFixture(Object testFixture) {
+        injector.injectMembers(testFixture);
+    }
+
+    public void cleanupMocks() {
+        for (Object mock : mocksToReset) {
+            Mockito.reset(mock);
+        }
+    }
+
+    public void cleanupInjector() {
+        injector.shutdown();
+    }
+
+    private LifecycleInjector createInjector(List<Module> modules, List<Module> overrides) {
+        return InjectorBuilder.fromModules(modules).overrideWith(overrides).createInjector();
+    }
+
+    private void getModulesForTestClass(Class<?> testClass) {
+        final List<Class<? extends Module>> moduleClasses = new ArrayList<>();
+
+        moduleClasses.addAll(getModulesForAnnotatedClass(testClass));
+        for (Class<?> parentClass : ClassUtils.getAllSuperclasses(testClass)) {
+            moduleClasses.addAll(getModulesForAnnotatedClass(parentClass));
+        }
+        for (Class<? extends Module> moduleClass : moduleClasses) {
+            try {
+                modulesForTestClass.add(moduleClass.newInstance());
+            } catch (InstantiationException | IllegalAccessException e) {
+                try {
+                    Constructor<?> constructor = moduleClass.getDeclaredConstructors()[0];
+                    constructor.setAccessible(true);
+                    modulesForTestClass.add((Module) constructor.newInstance());
+                } catch (Exception ex) {
+                    throw new RuntimeException("Error instantiating module " + moduleClass
+                            + ". Please ensure that the module is public and has a no-arg constructor", e);
+                }
+            }
+        }
+    }
+
+    private List<Class<? extends Module>> getModulesForAnnotatedClass(Class<?> testClass) {
+        final Annotation annotation = testClass.getAnnotation(ModulesForTesting.class);
+        if (annotation != null) {
+            return Arrays.asList(((ModulesForTesting) annotation).value());
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    private void getMocksForTestClass(Class<?> testClass) {
+        getMocksForAnnotatedFields(testClass);
+        for (Class<?> parentClass : ClassUtils.getAllSuperclasses(testClass)) {
+            getMocksForAnnotatedFields(parentClass);
+        }
+    }
+
+    @SuppressWarnings("rawtypes")
+    private void getMocksForAnnotatedFields(Class<?> testClass) {
+
+        for (Field field : testClass.getDeclaredFields()) {
+            if (field.isAnnotationPresent(ReplaceWithMock.class)) {
+                overrideModules.add(new MockitoOverrideModule(field.getAnnotation(ReplaceWithMock.class), field.getType()));
+            }
+            if (field.isAnnotationPresent(WrapWithSpy.class)) {
```

Comment:
It was not, just added a change for that. Right now I'm supporting the Named qualifier via the @WrapWithSpy(name="foo"). At some point we'll probably need to support any annotation qualifier, but I don't want to tackle that just yet.